File: src/main/java/com/team3/gudit/auth/redis/RefreshTokenCacheRepository.java
package com.team3.gudit.auth.redis;

import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.StringRedisTemplate;
import org.springframework.stereotype.Repository;

import java.time.Duration;
import java.util.Optional;

@Repository
@RequiredArgsConstructor
public class RefreshTokenCacheRepository {

    private final StringRedisTemplate redisTemplate;

    private static final String PREFIX = "refresh-token:";

    public void save(Long userId, String tokenHash, Duration ttl) {
        redisTemplate.opsForValue().set(PREFIX + userId, tokenHash, ttl);
    }

    public Optional<String> findByUserId(Long userId) {
        return Optional.ofNullable(
                redisTemplate.opsForValue()
                        .get(PREFIX + userId)
        );
    }

    public void delete(Long userId) {
        redisTemplate.delete(PREFIX + userId);
    }
}

File: src/main/kotlin/com/team3/gudit/auth/redis/RefreshTokenCacheRepository.kt
package com.team3.gudit.auth.redis

import org.springframework.data.redis.core.StringRedisTemplate
import org.springframework.stereotype.Repository
import java.time.Duration
import java.util.Optional

@Repository
class RefreshTokenCacheRepository(private val redisTemplate: StringRedisTemplate) {
    fun save(userId: Long, tokenHash: String, ttl: Duration) {
        redisTemplate.opsForValue().set(PREFIX + userId, tokenHash, ttl)
    }

    fun findByUserId(userId: Long): Optional<String> =
        Optional.ofNullable(redisTemplate.opsForValue().get(PREFIX + userId))

    fun delete(userId: Long) {
        redisTemplate.delete(PREFIX + userId)
    }

    companion object {
        private const val PREFIX = "refresh-token:"
    }
}
